### PR TITLE
crystal 0.36.0

### PR DESCRIPTION
**Pull request by bcardiff**
There is a new release of crystal and shards.

Support for llvm@11 is not production ready, so we stick with llvm@9 still.

### CODE REVIEW

File: Formula/crystal.rb [hunk 1]
```diff
@@ -5,12 +5,12 @@ class Crystal < Formula
   revision 2
```
Comment (Blacksmoke16) [inline Formula/crystal.rb:5]: Is this still needed given this is a new minor version?

**Comment by carlocab** [comment]
>Support for llvm@11 is not production ready, so we stick with llvm@9 still.

Will it build with llvm@11, though? If so, we may wish to use llvm@11 on ARM to get this bottled for ARM. We've done similar things with other formulae (e.g. using beta versions of Rust and Go on ARM.)

**Comment by bcardiff** [comment]
It will built, but if release mode is used the binary will fail to run. We are tracking at https://github.com/crystal-lang/crystal/issues/10220.

Using the compiler in non-release mode will be really slow.

**Comment by carlocab** [comment]
>Using the compiler in non-release mode will be really slow.

Is it slow enough to make it worse than not being able to use crystal at all? (Genuine question.)

**Comment by bcardiff** [comment]
Of course not :-), but I would not use llvm@11 non-arm platforms.

I don't object if there is something to be changed that will allow experimentation on arm, either here or in another PR. I don't have access to an M1 though.

**Comment by carlocab** [comment]
Well, if you wish, you can depend on `llvm` on ARM but `llvm@9` otherwise. You can use `if Hardware::CPU.arm?` for flow control.

Will the install block also need to be changed to accommodate the differing dependencies?

**Comment by carlocab** [comment]
I'm also curious about the `bdw-gc` resource block. The formula appears to make no use of it. It also declares the same version as the `bdw-gc` formula.

**Comment by bcardiff** [comment]
> You can use if Hardware::CPU.arm? for flow control.

Ok, but how it will boot? Crystal is bootstrapped. We use the previous compiler (downloaded from github) to bulid the latest release on boot. I guess we would still need a first compiler that will work natively on arm, right? Or there is some cross-compilation on behalf homebrew I am missing?

> Will the install block also need to be changed to accommodate the differing dependencies?

I don't think so.

> I'm also curious about the bdw-gc resource block. The formula appears to make no use of it.

It is used: `(prefix/"embedded/lib").install "#{buildpath/"gc"}/.libs/libgc.a"`.

It is the same version plus an unreleased upstream patch for coroutines support. Otherwise we would use the formula probably. Although, is nice to statically link the gc.

**Comment by carlocab** [comment]
>Ok, but how it will boot? Crystal is bootstrapped. We use the previous compiler (downloaded from github) to bulid the latest release on boot. I guess we would still need a first compiler that will work natively on arm, right?

It can probably run the previous compiler under Rosetta. I _think_ it should do this automatically. A natively compiled bootstrap compiler would be nice, though. (There are things Rosetta can't translate though. e.g. binaries compiled with AVX instructions. But I suspect your binaries don't have that, as that would make them not work on slightly older Intel processors.)

>It is the same version plus an unreleased upstream patch for coroutines support. Otherwise we would use the formula probably. Although, is nice to statically link the gc.

I missed that. I even did Ctrl-F `bdw-gc`; not sure why I didn't find it. I think I left parts of the install block unexpanded. This is fine then.

**Comment by deepj** [comment]
For me it would be good to have some experimental version of Crystal on ARM

**Review by carlocab — APPROVED** [review]
Let's merge this PR as is. We can try building for ARM in another one.

**Comment by carlocab** [comment]
See #69820.

**Comment by carlocab** [comment]
Ran into a wall with #69820, so I closed it. If any of you have any suggestions for what to do there, I'll be happy to reopen it.

**Pull request closed without merging**